### DISCUSSION TITLE
Observing any event loss with internal vector metrics

### DISCUSSION THREAD

**Discussion by LinTechSo** [vectordotdev/vector#18351] [Q&A] [answered]
Hi.
i have an question about what are the best metrics that we can rely on for observing vector event log loss at the sink .
any advice would be appreciated

**Comment by jszwedko** [comment, marked as answer]
Hi @LinTechSo ,

You can monitor `component_discarded_events_total{intentional=false}` for unintentionally discarded events.

**Reply by LinTechSo** [reply]
Hi. Thanks for the reply.
What about we configure the buffer to block instead of drop ? Is it related to the above metric ?
Thanks a lot.

**Reply by jszwedko** [reply]
If you configure the buffer to block instead of discard events then it will not discard events, but block instead. The buffers have their own set of metrics which you can find here: https://vector.dev/docs/reference/configuration/sources/internal_metrics/#buffer_byte_size (look for the ones beginning with `buffer_`). Hope this helps!
